react-components/src/hooks/scenes/use3dScenes.tsx

```diff
@@ -56,40 +56,58 @@ export const use3dScenes = (
   const fdmSdk = useMemo(() => new FdmSDK(sdk), [sdk]);
 
   const queryFunction: QueryFunction<Record<Space, Record<ExternalId, SceneData>>> = async () => {
-    const scenesQuery = createGetScenesQuery();
-
-    try {
-      const queryResult: Use3dScenesQueryResult = (
-        await fdmSdk.queryNodesAndEdges<
-          typeof scenesQuery,
-          [
-            { source: typeof SCENE_SOURCE; properties: SceneConfigurationProperties },
-            { source: typeof TRANSFORMATION_SOURCE; properties: Transformation3d },
-            { source: typeof ENVIRONMENT_MAP_SOURCE; properties: SkyboxProperties },
-            {
-              source: typeof IMAGE_360_COLLECTION_SOURCE;
-              properties: Cdf3dImage360CollectionProperties;
-            },
-            { source: typeof REVISION_SOURCE; properties: Cdf3dRevisionProperties },
-            { source: typeof GROUND_PLANE_SOURCE; properties: GroundPlaneProperties }
-          ]
-        >(scenesQuery)
-      ).items;
-
-      const scenesMap = createMapOfScenes(queryResult.scenes, queryResult.sceneSkybox);
-      populateSceneMapWithModels(queryResult.sceneModels, scenesMap);
-      populateSceneMapWith360Images(queryResult.scene360Collections, scenesMap);
-      populateSceneMapWithGroundplanes(
-        queryResult.sceneGroundPlanes,
-        queryResult.sceneGroundPlaneEdges,
-        scenesMap
-      );
-
-      return scenesMap;
-    } catch (error) {
-      console.warn("Scene space doesn't exist or has no scenes with 3D models");
-      return {};
+    const allScenes: Use3dScenesQueryResult = {
+      scenes: [],
+      sceneModels: [],
+      scene360Collections: [],
+      sceneGroundPlanes: [],
+      sceneGroundPlaneEdges: [],
+      sceneSkybox: []
+    };
+    let hasMore = true;
+    let cursor: string | undefined;
+    let scenesQuery = createGetScenesQuery();
+    const limit = 100;
+    const scenesLimit = scenesQuery.with.scenes.limit;
```

Comment:
Why do we initialize `sceneQuery` here when it's reinitialized right after inside the while loop after?

Maybe `sceneLimit` should instead be initialized from a constant?